File: backend/mobili-boot/src/main/java/com/mobili/backend/module/user/dto/ProfileDTO.java
package com.mobili.backend.module.user.dto;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProfileDTO {
    private Long id;
    private String firstname;
    private String lastname;
    private String login;
    private String email;
    private String avatarUrl;
    private boolean enabled;
    private Double balance;
    private List<String> roles;
    private Long partnerId;
    /** Compte gare (responsable d’une gare) */
    private Long stationId;
    private String stationName;
    /**
     * Rôle GARE : la gare est validée (booléen {@code Station.validated}) et active, donc
     * trajets, scanner, accès compagnie autorisés. {@code null} si pas gare.
     */
    private Boolean gareOperationsEnabled;
    private Integer totalBookingsCount;

    /** NONE, PENDING, APPROVED, REJECTED, EXPIRED — chauffeur covoiturage uniquement. */
    private String covoiturageKycStatus;
    /** Fin de validité de la CNI (ISO yyyy-MM-dd), si applicable. */
    private String covoiturageIdValidUntil;
    private String covoiturageVehicleBrand;
    private String covoiturageVehiclePlate;
    private String covoiturageVehicleColor;
    private String covoiturageGreyCardNumber;
    private String covoiturageVehiclePhotoUrl;
    /** Photo portrait du conducteur (KYC covoiturage). */
    private String covoiturageDriverPhotoUrl;
    /**
     * Jours avant expiration de l’ID (négatif si dépassée). null si non applicable.
     */
    private Long covoiturageKycDaysUntilExpiry;
    private Boolean covoiturageKycExpiringWithin30Days;
    private Boolean covoiturageKycIsDocumentExpired;
    /** Inscription covoiturage grand public (non rattaché à une compagnie Mobili). */
    private Boolean covoiturageSoloProfile;
}

File: mobile_app/lib/features/auth/domain/models/profile_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'profile_dto.g.dart';

@JsonSerializable()
class ProfileDto {
  const ProfileDto({
    required this.id,
    required this.firstname,
    required this.lastname,
    this.email,
    this.phone,
    required this.login,
    required this.roles,
    required this.enabled,
    this.avatarUrl,
    this.covoiturageKycStatus,
    this.covoiturageIdValidUntil,
    this.covoiturageVehicleBrand,
    this.covoiturageVehiclePlate,
    this.covoiturageVehicleColor,
    this.covoiturageGreyCardNumber,
    this.covoiturageVehiclePhotoUrl,
    this.covoiturageDriverPhotoUrl,
    this.covoiturageKycDaysUntilExpiry,
    this.covoiturageKycExpiringWithin30Days,
    this.covoiturageKycIsDocumentExpired,
    this.covoiturageSoloProfile,
  });

  final int id;
  final String firstname;
  final String lastname;
  final String? email; // ← optionnel
  final String? phone; // ← nouveau
  final String login;
  final String? avatarUrl;
  final List<String> roles;
  final bool enabled;

  // ── Covoiturage (conducteur particulier) ──────────────────────────────────
  final String? covoiturageKycStatus; // NONE | PENDING | APPROVED | REJECTED | EXPIRED
  final String? covoiturageIdValidUntil; // yyyy-MM-dd
  final String? covoiturageVehicleBrand;
  final String? covoiturageVehiclePlate;
  final String? covoiturageVehicleColor;
  final String? covoiturageGreyCardNumber;
  final String? covoiturageVehiclePhotoUrl;
  final String? covoiturageDriverPhotoUrl;
  final int? covoiturageKycDaysUntilExpiry;
  final bool? covoiturageKycExpiringWithin30Days;
  final bool? covoiturageKycIsDocumentExpired;
  final bool? covoiturageSoloProfile;

  // ---------------------------------------------------------------------------
  // Role helpers
  // ---------------------------------------------------------------------------

  bool get isUser => roles.contains('USER');
  bool get isPartner => roles.contains('PARTNER');
  bool get isGare => roles.contains('GARE');
  bool get isChauffeur => roles.contains('CHAUFFEUR');
  bool get isAdmin => roles.contains('ADMIN');

  String get fullName => '$firstname $lastname';

  // ---------------------------------------------------------------------------
  // Covoiturage helpers
  // ---------------------------------------------------------------------------

  /// True si l'utilisateur a déjà soumis un dossier conducteur covoiturage
  /// (quel que soit le statut KYC).
  bool get hasCovoiturageProfile =>
      covoiturageSoloProfile == true || covoiturageKycStatus != null;

  bool get covoiturageKycApproved => covoiturageKycStatus == 'APPROVED';
  bool get covoiturageKycPending => covoiturageKycStatus == 'PENDING';
  bool get covoiturageKycRejected => covoiturageKycStatus == 'REJECTED';
  bool get covoiturageKycExpired => covoiturageKycStatus == 'EXPIRED';

  // ---------------------------------------------------------------------------
  // Serialisation
  // ---------------------------------------------------------------------------

  factory ProfileDto.fromJson(Map<String, dynamic> json) =>
      _$ProfileDtoFromJson(json);

  Map<String, dynamic> toJson() => _$ProfileDtoToJson(this);

  @override
  String toString() => 'ProfileDto(id: $id, login: $login, roles: $roles)';
}
